src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +14,34 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException(e);
```

Comment:
Consider adding a descriptive message when throwing a new `RuntimeException` in the catch block. This will provide more context about the error, such as "Failed to add movie".